src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4122,7 +4122,23 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                     // can succeed.  This call will mount the drive if it wasn't already.
                     executionContext.SessionState.Drive.GetAtScope(wordToComplete.Substring(0, 1), "global");
                 }
+#if UNIX
+                StringBuilder sb = new StringBuilder();
+
+                foreach (var ch in wordToComplete.ToCharArray())
+                {
+                    if (Char.IsLetter(ch))
```

Comment:
I'm no expert in Unicode - but I can imagine there are letters that have no upper/lowercase variants, so I would avoid the wildcard if ToUpper == ToLower.

Reply:
German `ß` for example doesn't have upper in dotnet. (The correct capitalization is probably `SS`, but dotnet doesn't seem to do it, same as python 2.)
Also the The Turkish `I` is a classical example: if default locale is used, then lower-cased `I` will be different. We probably can skip these details as a first implementation.

Note that `[aa]` is a valid regex.

Reply:
Thanks! Fixed.